src/zhinst/toolkit/nodetree/nodetree.py

```diff
@@ -121,6 +121,25 @@ def add(self, node: t.Union[Node, str], value: t.Any) -> None:
         except AttributeError as exception:
             raise AttributeError("No set transaction is in progress.") from exception
 
+    def add_raw_list(self, node_value_pairs: t.List[t.Tuple[str, t.Any]]):
+        """Adds multiple set commands at a time.
+
+        Args:
+            node_value_pairs: List of settings in the form of
+            (node_string, value)
+
+        Note: settings can only take strings which to
+                describe a node, but no node objects
+        """
+        try:
+            self._queue += list(map(
+                lambda tup: (self._root.string_to_raw_path(tup[0]), tup[1]),
+                node_value_pairs
+            ))
```

Comment:
Why do you convert the string to a raw path here? The function name already states that it is a raw list here ... I would just trust (mention it in the docstring) that the paths are already correct. This simplifies the code drastically